modules/gapi/src/streaming/vpl/vpl_accel_policy.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef GAPI_VPL_ACCEL_POLICY_HPP
+#define GAPI_VPL_ACCEL_POLICY_HPP
+
+#include <functional>
+#include <memory>
+#include <type_traits>
+
+#include <opencv2/gapi/media.hpp>
+
+#ifdef HAVE_ONEVPL
+#include <vpl/mfxvideo.h>
+
+namespace cv {
+namespace gapi {
+namespace wip {
+
+class Surface;
+struct VPLAccelerationPolicy
+{
+    virtual ~VPLAccelerationPolicy() {}
+
+    using pool_key_t = void*;
+
+    using session_t = mfxSession;
+    using surface_t = Surface;
+    using surface_ptr_t = std::shared_ptr<surface_t>;
+    using surface_weak_ptr_t = std::weak_ptr<surface_t>;
+    using surface_ptr_ctr_t = std::function<surface_ptr_t(std::shared_ptr<void> out_buf_ptr,
+                                                          size_t out_buf_ptr_offset,
+                                                          size_t out_buf_ptr_size)>;
+
+    virtual void init(session_t session) = 0;
+    virtual void deinit(session_t session) = 0;
+
+    // Limitation: cannot give guarantee in succesful memory realloccation
+    // for existing workspace in existing pool (see realloc)
+    // thus it is not implemented,
+    // PLEASE provide initial memory area large enough
```

Comment:
Do you have appropriate asserts on that?

Reply:
it forbidden "by design" No methods for reallocating Pool memory...

In case of free-surface requests fails during runtime execution then WARNING would be printed and application pipeline would stuck before consumer releases captured MediaFrame (probably in another thread)

So  this is the place where 

    //TODO Configure preallocation size (how many frames we can hold)
    const size_t preallocated_frames_count = 30;


shots


As far i remember we discussed to make it configurable using compile_args